test/test-condvar.c

```diff
@@ -25,28 +25,55 @@
 #include <string.h>
 #include <errno.h>
 
-typedef struct {
+typedef struct worker_config {
   uv_mutex_t mutex;
-  uv_cond_t cond;
-  int delay;
+  uv_cond_t cond_1, cond_2;
+  int signal_delay, wait_delay;
   int use_broadcast;
-  volatile int posted;
+  volatile int posted_1, posted_2;
+  void (*signal_cond)(struct worker_config* c,
+                      uv_cond_t* cond,
+                      volatile int* flag);
```

Comment:
Using callbacks seems a little overcomplicated; `.signal_cond` looks like it's always the same function.